### ISSUE TITLE
Zero denominator causes RejectOptionClassification.fit() to fail

### ISSUE THREAD

**Issue by gabastil** [Trusted-AI/AIF360#307]
In the case of a zero in the denominator of the rate functions (e.g., `true_negative_rate`), the fit function throws an error.

In my case, the `classified_transf_metric.true_negative_rate()` function in lines 143 and 144 of `aif360/algorithms/postprocessing/reject_option_classification.py` causes the entire expression to return as `nan` as it returns `nan` in the addition operation that is assigned to `balanced_acc_arr[cnt]`.

The above, in turn, causes downstream operations to throw errors.

I propose returning zero and sending a warning message that this has occurred, but not sure if that is the best route forward for this issue.

<img width="812" alt="image" src="https://user-images.githubusercontent.com/7361746/166909008-21df460c-0d76-4ad9-a648-7456532b097a.png">

**Comment by krvarshney** [comment]
I don't think that returning 0 when there is a divide by zero error is the correct solution. https://en.wikipedia.org/wiki/Division_by_zero

**Comment by hoffmansc** [comment]
It looks like this is what sklearn does (e.g. [recall_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html)) and what we do in the [sklearn-compatible metrics](https://github.com/Trusted-AI/AIF360/blob/master/aif360/sklearn/metrics/metrics.py#L167-L170) as well.

However, it seems like a really breaking issue if your dataset has no positive or no negative samples at all. What's the point of running a debiasing algorithm on such a dataset?

**Comment by gabastil** [comment]
The dataset does have either positive or negative samples, but I see where you are coming from.

Regardless, even if it's "failing gracefully" like a helpful message describing the error (i.e., a division by zero error at point of division), this issue needs to be addressed.
